Source/RxSwift/mergeWith.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  mergeWith.swift
+//  RxSwiftExt
+//
+//  Created by Joan Disho on 12.05.18.
+//  Copyright © 2018 RxSwiftCommunity. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+extension Observable {
+    /**
+     Merges `self` with another observable.
+
+     - parameter other: An other observable.
+     - returns: A merged observable.
+     */
+    public func merge(with other: Observable<E>) -> Observable<E> {
```

Comment:
What do you think making `other` a Variadic argument ? That way it could be used to merge any number of Observables of type E. 

That could look like this: 

```swift
public func merge(with others: Observable<E>...) -> Observable<E> {
    return Observable.merge([self] + others)
}
```